src/v/utils/log_hist.h

```diff
@@ -0,0 +1,184 @@
+    /// and the recording will simply be ignored.
+    class measurement : public boost::intrusive::list_base_hook<> {
+    public:
+        explicit measurement(log_hist& h)
+          : _h(std::ref(h))
+          , _begin_t(log_hist::clock_type::now()) {
+            _h.get()._probes.push_back(*this);
+        }
+        measurement(const measurement&) = delete;
+        measurement& operator=(const measurement&) = delete;
+        measurement(measurement&& o) noexcept
+          : _detached(o._detached)
+          , _trace(o._trace)
+          , _h(o._h)
+          , _begin_t(o._begin_t) {
+            o.detach_hdr_hist();
+        }
+        measurement& operator=(measurement&& o) noexcept {
+            if (this != &o) {
+                this->~measurement();
+                new (this) measurement(std::move(o));
+            }
+            return *this;
+        }
+        ~measurement() noexcept {
+            if (!_detached) {
+                _h.get()._probes.erase(_h.get()._probes.iterator_to(*this));
+                // !detached && trace
+                // do not move outside of this nested if
+                if (_trace) {
+                    _h.get().record(compute_duration());
+                }
+            }
+        }
+
+        void set_trace(bool b) { _trace = b; }
+
+    private:
+        friend log_hist;
+
+        int64_t compute_duration() const {
+            return std::chrono::duration_cast<duration_t>(
+                     log_hist::clock_type::now() - _begin_t)
+              .count();
+        }
+
+        void detach_hdr_hist() { _detached = true; }
+
+        bool _detached = false;
+        bool _trace = true;
+        std::reference_wrapper<log_hist> _h;
+        log_hist::clock_type::time_point _begin_t;
+
+        friend std::ostream& operator<<(std::ostream& o, const measurement&);
+    };
+
+    std::unique_ptr<measurement> auto_measure() {
+        return std::make_unique<measurement>(*this);
+    }
+
+    log_hist()
+      : _counts(number_of_buckets) {}
+
+    /*
+     * record expects values of that are equivalent to `duration_t::count()`
+     * so make sure the input is scaled correctly.
+     */
+    void record(uint64_t val) {
+        _sample_sum += val;
+        const int i = std::min(
+          std::max(first_bucket_clz - std::countl_zero(val), 0),
+          (int)_counts.size() - 1);
+        _counts[i]++;
+    }
+
+    void record(std::unique_ptr<measurement> m) {
+        record(m->compute_duration());
+    }
+
+    seastar::metrics::histogram seastar_histogram_logform(double scale) const {
+        seastar::metrics::histogram hist;
+        hist.buckets.resize(_counts.size());
+        hist.sample_sum = static_cast<double>(_sample_sum) / scale;
+
+        uint64_t cumulative_count = 0;
+        for (size_t i = 0; i < _counts.size(); i++) {
+            auto& bucket = hist.buckets[i];
+
+            cumulative_count += _counts[i];
+            bucket.count += cumulative_count;
```

Comment:
why is this `+=` ? Do we ever expect the value to have an existing value somehow? Seems like it should be `=`.

Reply:
Nice catch, switch to `=`.